src/frontend/magnify/src/context/auth/index.tsx

```diff
@@ -0,0 +1,50 @@
+import { useQueryClient } from '@tanstack/react-query';
+import React, { useEffect, useState } from 'react';
+import { Maybe } from 'yup/es/types';
+import { User } from '../../types/entities/user';
+import { MagnifyQueryKeys } from '../../utils/constants/react-query';
+
+export interface AuthContextInterface {
+  user: Maybe<User> | null;
+  updateUser: (user: Maybe<User> | null) => void;
+}
+
+const AuthContext = React.createContext<AuthContextInterface | undefined>(undefined);
+
+type AuthContextProviderProps = {
+  children: React.ReactNode;
+  initialUser?: Maybe<User> | null;
+};
+
+export const AuthContextProvider = ({ children, ...props }: AuthContextProviderProps) => {
+  const [user, setUser] = useState<Maybe<User> | null>(props.initialUser ?? undefined);
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    if (props.initialUser != null) {
+      queryClient.setQueryData([MagnifyQueryKeys.AUTH_USER], props.initialUser);
+    }
+  }, []);
+
+  const context: AuthContextInterface = React.useMemo(
+    () => ({
+      user: user,
+      updateUser: (user: Maybe<User> | null) => {
+        queryClient.setQueryData([MagnifyQueryKeys.AUTH_USER], user);
+        setUser(user);
+      },
+    }),
+    [user],
+  );
+  return <AuthContext.Provider value={context}>{children}</AuthContext.Provider>;
+};
+
+export const useAuthContext = () => {
```

Comment:
Not declaring the context like `const AuthContext = React.createContext<AuthContextInterface | undefined>(undefined);` but with `const AuthContext = React.createContext<AuthContextInterface>({} as any);` would have saved you the need to create a ad-hoc function like this because React it-self would have trigger the error below if needed ( `throw new Error(`useAuthContext must be used within a AuthContextProvider`);`)